swiftsimio/writer.py

```diff
@@ -333,9 +336,9 @@ def _write_metadata(self, handle: h5py.File, names_to_write: List):
         Writes metadata to file based on the information passed to the object
         and the information in the particle groups.
         """
-
-        number_of_particles = [0] * 6
-        mass_table = [0.0] * 6
+        part_types = max(metadata.particle.particle_name_underscores.keys())+1
```

Comment:
Is this okay within SWIFT? I don't like the idea of doing this at all...

Reply:
Whats your concern here? I just tested an input i created that had 1 gas particle, 6 particles of type 6  (boundary) and 1 of type 7 (fluid). SWIFT opened the file fine, and imported 1 gas particle (as expected) before giving up in space_regrid due to not enough particles. Anything that is past the end of `swift_type_count` is ignored. For our work we'll probably want to extend/override this but also not remove already reserved part names, so adding on boundary and fluid particles as new types works. This was the only way I could think of extending the number of allowed types without manually just changing the value of 6.

Reply:
Oh I had just assumed that using any number of items here apart from 6 would kill SWIFT -- if this works then I'm happy to go with this.